VocaDbWeb/Scripts/Components/VdbPlayer/EmbedNiconico.tsx

```diff
@@ -0,0 +1,311 @@
+				JSON.parse(JSON.stringify(pv)),
+			);
+
+			await this.attach();
+
+			this.assertPlayerAttached();
+			if (!this.player) return;
+
+			// Wait for iframe to load.
+			this.player.onload = (): void => {
+				this.assertPlayerAttached();
+				if (!this.player) return;
+
+				this.player.onload = null;
+
+				VdbPlayerConsole.debug('Niconico iframe loaded');
+
+				resolve();
+			};
+
+			this.player.src = `https://embed.nicovideo.jp/watch/${
+				pv.pvId
+			}?${qs.stringify({
+				jsapi: 1,
+				playerId: 1,
+			})}`;
+		});
+	};
+
+	// Code from: https://blog.hayu.io/web/create/nicovideo-embed-player-api/.
+	private postMessage = (message: any): void => {
+		this.assertPlayerAttached();
+		if (!this.player) return;
+
+		this.player.contentWindow?.postMessage(
+			{
+				...message,
+				playerId: '1' /* Needs to be a string, not a number. */,
+				sourceConnectorType: 1,
+			},
+			PVPlayerNiconico.origin,
+		);
+	};
+
+	public play = (): void => {
+		VdbPlayerConsole.debug('PVPlayerNiconico.play');
+
+		this.postMessage({ eventName: 'play' });
+	};
+
+	public pause = (): void => {
+		VdbPlayerConsole.debug('PVPlayerNiconico.pause');
+
+		this.postMessage({ eventName: 'pause' });
+	};
+
+	public seekTo = (seconds: number): void => {
+		VdbPlayerConsole.debug('PVPlayerNiconico.seekTo');
+
+		this.postMessage({ eventName: 'seek', data: { time: seconds * 1000 } });
+	};
+
+	public mute = (): void => {
+		VdbPlayerConsole.debug('PVPlayerNiconico.mute');
+
+		this.postMessage({
+			eventName: 'mute',
+			data: { mute: true },
+		});
+	};
+
+	public unmute = (): void => {
+		VdbPlayerConsole.debug('PVPlayerNiconico.unmute');
+
+		this.postMessage({
+			eventName: 'mute',
+			data: { mute: false },
+		});
+	};
+}
+
+interface EmbedNiconicoProps extends IPVPlayerOptions {
+	playerRef: React.MutableRefObject<IPVPlayer | undefined>;
+}
+
+const EmbedNiconico = React.memo(
+	({ playerRef, ...options }: EmbedNiconicoProps): React.ReactElement => {
+		VdbPlayerConsole.debug('EmbedNiconico');
+
+		const playerElementRef = React.useRef<HTMLIFrameElement>(undefined!);
+
+		React.useEffect(() => {
+			playerRef.current = new PVPlayerNiconico(playerElementRef, options);
+		}, [playerRef, options]);
+
+		return (
+			<div style={{ width: '100%', height: '100%' }}>
+				{/* eslint-disable-next-line jsx-a11y/iframe-has-title */}
+				<iframe
+					ref={playerElementRef}
+					width="100%"
+					height="100%"
+					allowFullScreen
+					style={{ border: 'none' }}
+					// The player has to have the allow="autoplay" attribute.
+					// Otherwise it throws a NotAllowedError: "play() failed because the user didn't interact with the document first".
+					// See also: https://github.com/vimeo/player.js/issues/389.
+					// NOTE: An iframe element created by `PVPlayerNiconico.playerFactory.create` doesn't have the allow="autoplay" attribute,
+					// which causes the above issue when trying to autoplay a video.
+					allow="autoplay; fullscreen"

```

Comment:
The player has to have the `allow="autoplay"` attribute. Otherwise it throws a `NotAllowedError: "play() failed because the user didn't interact with the document first"`. See also: https://github.com/vimeo/player.js/issues/389.
NOTE: An iframe element created by `PVPlayerNiconico.playerFactory.create` doesn't have the `allow="autoplay" attribute`, which causes the above issue when trying to autoplay a video.